test/ruby/test_gc.rb

```diff
@@ -262,9 +262,12 @@ def test_latest_gc_info_need_major_by
       objects.append(100.times.map { '*' })
     end
 
-    assert_not_nil GC.latest_gc_info(:need_major_by)
+    need_major_by = GC.latest_gc_info(:need_major_by)
```

Comment:
I appreciate the git log message indicating why the change was made, but I think a comment here would make it less likely someone accidentally makes it flaky again.